Frontend/student-hub/src/pages/MentorshipPage.js

```diff
@@ -20,55 +22,224 @@ const SearchContainer = styled(Paper)({
         setError('Failed to fetch mentorships');
       }
+      setIsLoading(false)
     };
-
     fetchMentorships();
   }, [setError]);
 
+  // fetch requests received
+  useEffect(() => {
+    const fetchMentorshipRequest = async () => {
+      try {
+        const { token } = getUserIDToken();
+        const response = await fetch(`${process.env.REACT_APP_API}/mentorship/mentor-requests`, {
+          method: 'GET',
+          headers: {
+            Authorization: `Bearer ${token}`,
+          },
+        });
+
+        if (!response.ok) {
+          setError('Failed to fetch mentorship');
+        }
+        const requestData = await response.json();
+        setMentorshipRequest(requestData);
+      }
+      catch (error) {
+        setError('Fetch mentorshp requests error:', error)
+      }
+    };
+    fetchMentorshipRequest();
+  }, [setError])
+
   //search
   const handleSearch = (event) => {
     setSearchTerm(event.target.value);
+  };
 
+  //request mentor
+  const handleRequestMentor = async () => {
+    try {
+      const { token } = getUserIDToken();
+      const response = await fetch(`${process.env.REACT_APP_API}/mentorship/request-mentor`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: `Bearer ${token}`
+        },
+        body: JSON.stringify({ userId: selectedMentorId, note })
+      });
+
+      if (!response.ok) {
+        setError('Failed to fetch mentorship');
+      }
+      alert('Mentor request sent successfuly');
+      handleCloseRequestDialog();
+    } catch (error) {
+      setError('Error requesting mentor:', error)
+    }
   };
 
+  // respond to a mentorship request.
+  const handleRespondMentorship = async (userId,mentorId, status) => {
+    try{
+      const { token } = getUserIDToken();
+      const response = await fetch(`${process.env.REACT_APP_API}/mentorship/respond-mentorship`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: `Bearer ${token}`,
+        },
+        body: JSON.stringify({ userId,mentorId, status})
+      });
+
+      if (!response.ok) {
+        setError('Failed to fetch mentorship');
+      }
+      const requestData = await response.json();
+      setMentorshipRequest(requestData);
+      alert(`Mentorship request ${status}`);
+      setMentorshipRequest(mentorshipRequest.filter((request) => request.id !== userId))
+    }
+    catch (error){
+      setError('Respond mentorship error', error)
+    }
+  };
+
+  // fetch mentees information for a mentor
+  const fetchMentees = async () => {
+    try{
+      const { userId, token } = getUserIDToken();
+
+      const response = await fetch(`${process.env.REACT_APP_API}/mentorship/${userId}/mentees`, {
+        method: 'GET',
+        headers: {
+          Authorization: `Bearer ${token}`
+        }
+      });
+      const data = await response.json();
+        setFilteredMenteeList(data)
```

Comment:
if data here has length > 0 then you should call `setUserRole('Mentor')` -- is this the source-of-truth for what we defined as a Mentor (a user with mentees? or is there a separate flag on the User object?)